neptune/new/internal/operation_processors/async_operation_processor.py

```diff
@@ -89,24 +90,59 @@ def flush(self):
     def start(self):
         self._consumer.start()
 
+    def _wait_for_queue_empty(self, initial_queue_size: int, seconds: Optional[float]):
+        waiting_start = monotonic()
+        max_wait_time = self.STOP_QUEUE_MAX_TIME_NO_CONNECTION_SECONDS if seconds is None else seconds
+        if initial_queue_size > 0:
+            if self._consumer.last_backoff_time > 0:
+                click.echo(f"We have been experiencing connection interruptions during your run. "
+                           f"Client will now try to resume connection and sync data for the next "
+                           f"{max_wait_time} seconds. "
+                           f"You can also synchronize manually using neptune sync command.",
+                           sys.stderr)
+            click.echo(f"Waiting for the remaining {initial_queue_size} operations to synchronize with Neptune. "
+                       f"Do not kill you application.",
+                       sys.stderr)
+        if seconds is not None:
+            return self._queue.wait_for_empty(seconds)
```

Comment:
I think we should still print progress messages if seconds is not None.

Reply:
I kinda assumed that not-None seconds would be <10, but fair point